test/unit/slice_generator_test.rb

```diff
@@ -174,4 +174,24 @@ def fact_names
     assert_not_nil(actual_profile = actual_host['system_profile'])
     assert_equal 1024, actual_profile['system_memory_bytes']
   end
+
+  test 'reports an account for hosts with multiple pools' do
+    first_pool = @host.subscription_facet.pools.first
+    second_pool = FactoryBot.create(:katello_pool, account_number: nil, cp_id: 2)
+    @host.subscription_facet.pools = []
+    @host.subscription_facet.save!
+    @host.subscription_facet.pools << first_pool
+    @host.subscription_facet.pools << second_pool
+
+    batch = Host.where(id: @host.id).in_batches.first
+    generator = ForemanInventoryUpload::Generators::Slice.new(batch, [], 'slice_123')
+    
```

Comment:
nit, rubocop is angry about this: `Layout/TrailingWhitespace: Trailing whitespace detected.`